python/ray/serve/multiplex.py

```diff
@@ -90,8 +90,57 @@ def __init__(
         # Whether to push the multiplexed replica info to the controller.
         self._push_multiplexed_replica_info: bool = False
 
-        # Push the model IDs to the controller periodically.
-        run_background_task(self._push_model_ids())
+        # Model cache lock to ensure that only one model is loading/unloading at a time.
+        self._model_cache_lock = asyncio.Lock()
+        # The set of model IDs that are being loaded. This is used to early push
+        # model ids info to the controller. The tasks will be added when there is cache
+        # miss, and will be removed when the model is loaded successfully or
+        # failed to load.
+        self._model_load_tasks: Set[str] = set()
+
+        self.metrics_pusher = MetricsPusher()
+        self.metrics_pusher.register_task(
+            self._push_model_ids_info,
+            PUSH_MULTIPLEXED_MODEL_IDS_INTERVAL_S,
+        )
+        self.metrics_pusher.start()
+
+    def _get_loading_and_loaded_model_ids(self) -> List[str]:
+        """Get the model IDs of the loaded models & loading models in the replica.
```

Comment:
nit: follow google style guide for docstrings. should start with a single-line description